Changelog November 2020 update (#47898)
CleverRaven/Cataclysm-DDA: data/changelog.txt

Require grenades and other explosives to be wielded before activation.
Add bigger mini-maps to all sidebar layouts.
Add keycode support to uilist.
Allow disabling robots by 'e'xamining them.
Warn if friendly creature or non-hostile NPC enters line of fire.

> Replaces the CAL/kg row of the (E)at menu with a more meaningful Satiety row.

Reviewer: ```suggestion
Replaces the CAL/kg column of the (E)at menu with a more meaningful Satiety column.
```

> Add a Mod to use MSX Characters and monster with the Ultica tileset

Reviewer: ```suggestion
Adds a Mod to use MSX characters and monsters with the Ultica tileset.
```
Still is not quite clear to me - will take a look at original PR.

Show missing proficiency failure rates for crafting.
Display contained items like 'water (plastic bottle)' instead of 'plastic bottle of water'.
Persist preferred aiming mode.

> Make Chibi Ultica a Stand-alone tileset
> Bring back MSX+

Reviewer: ```suggestion
Makes Chibi Ultica a stand-alone tileset.
Brings back MSX+.
```

Polymorph spells to transform your foes.
Spells using components system.
Aftershock: Add autodoc tiers and augmentation clinics.
Adds location variants with dinosaurs.
Aftershock: Megacity oriented overmap generation adjustments

> Aftershock: New Melee oriented bionics

Reviewer: ```suggestion
Aftershock: Adds new Melee oriented bionics.
```

Adds Acrocanthosaurus, dino baby overhaul.

> Add New Black Dragon Mutation Line.
> Adjust fog frequency in black dragon lair, minor map adjustments.

Reviewer: ```suggestion
Adds New Black Dragon Mutation Line.
Adjusts fog frequency in black dragon lair plus other (minor) map adjustments.
```

Adds monsters 'Scutellosaurus', 'Maiasaura', and 'Coelophysis'.
[Magiclysm] Attunements - Special abilities gained by mastering two schools of magic.

> Add monsters 'Pachyrhinosaurus', 'Pentaceratops', and 'Torosaurus'
> Organizes dinos into three main factions with defined behavior

Reviewer: ```suggestion
Adds monsters 'Pachyrhinosaurus', 'Pentaceratops', and 'Torosaurus'.
Organizes dinos into three main factions with defined inter-faction behavior.
```

Rebalance Jurassic spawns and monsters for realism, fix predator in-fighting.
Cap monsters at 1000 kg so butchering kind of works.
[Magiclysm] implement summonable motorcycle

> [Magiclysm] Adds a new house with technomancer themed loot.
> Aftershock: Aftershock: Replace Vanilla autodocs with Aftershock Average Autodocs
> Adds monsters 'Brontosaurus', 'Alamosaurus', and 'Brachiosaurus'
> [magiclysm] rings of wizardry, mojocycle tweaks
> [magiclysm] add bullet armor to magic rings and bracers of protection
> Convert 'dinoexhibit' to use templates, add roof, add baby dino variant
> Ankylosaurs Abound: Dyoplosaurus Nodosaurus Edmontonia

Reviewer: ```suggestion
[Magiclysm] Adds a new house with technomancer-themed loot.
Aftershock: Replaces Vanilla autodocs with Aftershock Average Autodocs.
DinoMod: Adds monsters 'Brontosaurus', 'Alamosaurus', and 'Brachiosaurus'.
[Magiclysm] Rings of wizardry and mojocycle tweaks
[Magiclysm] Adds bullet armor to magic rings and bracers of protection.
DinoMod: Converts 'dinoexhibit' to use templates, adds a roof, and adds a baby dino variant.
Ankylosaurs Abound: Dyoplosaurus Nodosaurus Edmontonia
```

NPCs to use ranged weapons against all enemies, and automatic fire against dangerous enemies.
Antifungal drugs work over time.
Don't damage hands when smashing but use your most armored part instead.
Guarantee at least one transversable path in the collapsed tower basement.
Make Solar Sensitivity mutation respect clothing coverage, precious.
Unlock NPC recruiting when you build the faction camp radio console.

> Clean_water recipe with purification tablets or charcoal water purifier.

Reviewer: ```suggestion
Fixed clean_water recipes with purification tablets and charcoal water purifier.
```
Or did it add them?

> Bring reloading components close to reality, make brass spawn in packs, make gunpowder units 100 mg

Reviewer: ```suggestion
Brings reloading components closer to reality, make brass casings spawn in packs, and makes gunpowder units 100 mg.
```
"Casings" may need double-checking.

Fix NPC dying immediately on accepting inhaler mission
Fix crash when attempting to butcher a corpse.
Fix gear getting damaged when weather has no damaging effects.
Improve invalidation of max_populated_zlev cache, fixes lighting in some teleportation scenarios.
Prevent dodging or blocking while insensible.

> Fix apparent light asymmetry when sun shines through a single-tile hole in the roof.

Reviewer: ```suggestion
Fixes apparent light asymmetry when the sun shines through a single-tile hole in the roof.
```

Made installation time mandatory for gunmods.

> Fix #44421 tents and vehicles are not blocking sunlight.

Reviewer: ```suggestion
Makes tents and vehicles block sunlight.
```

Fix scent map saved with invalid utf8 sequence.

> Prevent monsters from bashing obstacles based on tracking sounds.

Reviewer: This is a bit unclear. "based on" is meaning "only due to", or what?

Author: Only due to, i.e. they won't bash things unless they see or smell something, I agree it needs clarification, it's a pretty big change.
```suggestion
Monsters no longer bash obstacles based on sound. This prevents groups of monsters from getting stuck in a loop of bashing based on the sound of other monsters bashing.
```

Picking up an item stack fills up your pockets if it doesn't fit into one pocket.
Wielding combines item stacks if possible.
Robots can no longer be disabled through melee hits.

> Fix auto consume zones in vehicles.

Reviewer: ```suggestion
Fixes auto-consume zones in vehicles.
```

## Performance:
Add option to enable batching when it is available in SDL.
Skip unnecessary weather calculations for root cellar.
Cache drop_invalid_inventory and weight_carried.
Cache clairvoyance field.
Performance improvements related to map::process_fields.
Overhaul shadowcasting to be iterative instead of recursive.
Use plf linked list impl for shadowcasting.
Improve string_id comparison performance, add unit test and benchmarks.

> Improve rendering speed by optimizing find_tile_looks_like method (FPS improved ≈15%)

Reviewer: ```suggestion
Improves rendering speed by optimizing find_tile_looks_like method (FPS improved ~15%)
```
Make pure ASCII.

Greatly reduce overhead of processing idle vehicles.

> Optimize Creature::has_effect (performance improved ≈9.5%)
> Add a specialization for static/interned string_ids  (performance improved ≈30%))

Reviewer: ```suggestion
Optimizes Creature::has_effect (performance improved ~9.5%)
Adds a specialization for static/interned string_ids  (performance improved ~30%))
```

Unblacklist most remaining blacklisted mods for automated consistency checking.
Report log messages for failed tests.
Reinstate blueprint autocalc.
Refactored vision_test. Introduced map_test_case framework to build similar tests easier.
Add tests for better_pocket and best_pocket.

> Rename skills in preparation of some changes; merge speech & barter.

Reviewer: ```suggestion
Renames skills in preparation for some future improvements; merges speech & barter.
```

> Merge trapping and lockpicking skills into a new 'devices' skill, expand scope somewhat.

Reviewer: ```suggestion
Merges trapping and lockpicking skills into a new 'devices' skill, and expands its scope somewhat.
```

compose.py: allow default sized tilesheets to specify offsets.
Enable compiling an Android build under Windows.
Tools: add a Python script to reformat vehicles to use parts arrays.
Suppress Homebrew update for faster OSX builds.

> Add problem matchers for json errors, compiler errors, DebugLog emission and Catch2 errors.

Reviewer: ```suggestion
Adds problem matchers for json errors, compiler errors, DebugLog emissions, and Catch2 errors.
```

Add check enforcing almost-never-auto convention.

> Add a GitHub workflow running flake8 on the Python scripts.

Reviewer: ```suggestion
Adds a GitHub workflow running flake8 to check the Python scripts.
```

## I18N and A11Y:

> Fix translation of martial art initate message and computer translation.

Reviewer: ```suggestion
Fix translation of martial art initiate message and computer translation.
```
